### ISSUE TITLE
[Analyzer] Address Overflow in substract 00000000 - 0x1

### ISSUE THREAD

**Issue by lab313ru** [NationalSecurityAgency/ghidra#1560]
To reproduce:
1.  Load any binary, don't start analysis
2.  Create uninitialized segment at the offset 0x00000000
3.  Create a label at offset 0 named "*m*"
4.  Load *generic_clib* (it must contain *crypto.h/functions/m* with some function definition) into a project
5.  Start analysis (press A, then Analyze)
6.  Wait until the analyzer will start fdefs applying
7.  Get an exception

![изображение](https://user-images.githubusercontent.com/7189309/75096798-20252180-55b4-11ea-8ee6-74954f53310e.png)
![изображение](https://user-images.githubusercontent.com/7189309/75096816-55ca0a80-55b4-11ea-85f3-4d5a3903a508.png)
![изображение](https://user-images.githubusercontent.com/7189309/75096823-6bd7cb00-55b4-11ea-9c37-f6e3ea86bb3d.png)

**Comment by Marc-Bender** [comment]
got the same thing here on disassemby of a atmel avr firmware blob that does not use crypto at all. so i guess that crypto is not the issue here (or even necessary to mention)

just for completeness let me paste the details that @lab313ru kindly ommitted:

```
Analysis Task: GCC Exception Handlers - Address Overflow in subtract: code:000000 - 0x1
ghidra.program.model.address.AddressOutOfBoundsException: Address Overflow in subtract: code:000000 - 0x1
	at ghidra.program.model.address.AbstractAddressSpace.add(AbstractAddressSpace.java:493)
	at ghidra.program.model.address.GenericAddressSpace.add(GenericAddressSpace.java:21)
	at ghidra.program.model.address.GenericAddress.add(GenericAddress.java:237)
	at ghidra.app.plugin.exceptionhandlers.gcc.structures.ehFrame.FrameDescriptionEntry.createPcRange(FrameDescriptionEntry.java:308)
	at ghidra.app.plugin.exceptionhandlers.gcc.structures.ehFrame.FrameDescriptionEntry.create(FrameDescriptionEntry.java:468)
	at ghidra.app.plugin.exceptionhandlers.gcc.sections.DebugFrameSection.analyzeSection(DebugFrameSection.java:131)
	at ghidra.app.plugin.exceptionhandlers.gcc.sections.DebugFrameSection.analyze(DebugFrameSection.java:77)
	at ghidra.app.plugin.exceptionhandlers.gcc.GccExceptionAnalyzer.handleDebugFrameSection(GccExceptionAnalyzer.java:399)
	at ghidra.app.plugin.exceptionhandlers.gcc.GccExceptionAnalyzer.added(GccExceptionAnalyzer.java:145)
	at ghidra.app.plugin.core.analysis.AnalysisScheduler.runAnalyzer(AnalysisScheduler.java:185)
	at ghidra.app.plugin.core.analysis.AnalysisTask.applyTo(AnalysisTask.java:39)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager$AnalysisTaskWrapper.run(AutoAnalysisManager.java:685)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:785)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:664)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:629)
	at ghidra.app.plugin.core.analysis.AnalysisBackgroundCommand.applyTo(AnalysisBackgroundCommand.java:62)
	at ghidra.framework.plugintool.mgr.BackgroundCommandTask.run(BackgroundCommandTask.java:101)
	at ghidra.framework.plugintool.mgr.ToolTaskManager.run(ToolTaskManager.java:315)
	at java.base/java.lang.Thread.run(Thread.java:832)

---------------------------------------------------
Build Date: 2020-Feb-12 1149 EST
Ghidra Version: 9.1.2
Java Home: C:\Program Files\Java\jdk-14.0.1
JVM Version: Oracle Corporation 14.0.1
OS: Windows 10 10.0 amd64
Workstation: 192.168.0.3

```

**Comment by cyber-wizard-sedridor** [comment]
Just wanted to pop in and say this is still an issue on the latest version of Ghidra.

I ran into this issue when defining a new pspec file that had a memory block starting at 0x00000000. Upon analysis, the Apply Data Archives task crashed with the same error listed above, attempting to subtract 0x1 from the memory block's starting address (in this case, 00000000, making the task impossible).

While this issue most likely never presents itself during traditional binary analysis, it isn't out of the question to run into a 0x00000000 memory block while doing firmware analysis, where I ran into the issue.

As a short term fix, if you're just looking to suppress the error, you can remove the offending 0x00000000 memory block definition

**Comment by mumbel** [comment]
Running into this as well with ELF .o files loaded batch from an .a file.

In my case at least it is attempting to make a range with wrap, but def something up with the logic for a displacement of -1 to handle the wrap and it hits this un-handled exception.  My trace looks similar to @Marc-Bender , where it's something like: 
 `val=readInt(addr) ; createPcRange(val - 1);`

**Comment by pabx06** [comment]
What i do to avoid batch loading ar archive 
- gcc -shared *.o -o lib.so
- ld - r --whole-archive *.o --no-whole-archive -o lib.so

If you are lucky enough it would work

**Comment by emteere** [comment]
Seems related to another bug I'm fixing.  I'll take a look and fix it.
The FDE exception seems odd

**Comment by emteere** [comment]
@mumbel do you have a sample .o file?
Looking at the code, it would have to have an ehframe Frame Descriptor Entry record that refers to the first address and the .o is loaded at 0x0.
It is an easy fix to stop, I'd just like to make sure we do the right thing with it.

**Comment by mumbel** [comment]
@emteere   I was trying this out https://github.com/pine64/bl_iot_sdk/tree/master/components/network/ble/blecontroller/lib

I did find an issue in my opinion file though, so you would need that change most likely (2nd commit in my RISCV update PR)

**Comment by emteere** [comment]
This and and some associated should be fixed in 9.2 and master soon.
Thanks for the test file

**Comment by mumbel** [comment]
fixed by 71b50a6929b0a0c92ed9bfc33f32f0cc03cc17e4  ?

**Comment by emteere** [comment]
Fixed with e0196165070c264d836d386f6977a681c9df9d2e.

Fixed issues in Dwarf zero size symbol entries, pattern starts at 0x0, applied function signatures at 0x0, and after="function" tag in start patterns

**Comment by emteere** [comment]
Fixed in 9.2 and master.  Closing.  Please reopen if not fixed.
